variables.tf

```diff
@@ -2,10 +2,218 @@
+  validation {
+    condition = anytrue([
+      var.redis_version == null,
+      var.redis_version == "5",
+      var.redis_version == "6",
+      var.redis_version == "6.2"
+    ])
+    error_message = "Version must be 5, 6 or 6.2. If no value passed, the current ICD preferred version is used."
+  }
+}
+
+variable "region" {
+  description = "The region where you want to deploy your instance."
+  type        = string
+  default     = "us-south"
+}
+
+variable "configuration" {
+  description = "Database Configuration. Default values will get picked up if not all the values are passed."
+  type = object({
+    maxmemory                   = optional(number)
+    maxmemory-policy            = optional(string)
+    appendonly                  = optional(string)
+    maxmemory-samples           = optional(number)
+    stop-writes-on-bgsave-error = optional(string)
+  })
+  default = null
+}
+
+variable "cpu_count" {
+  description = "Allocated dedicated CPU per member. For shared CPU, set to 0. For more information, see https://cloud.ibm.com/docs/databases-for-redis?topic=databases-for-redis-resources-scaling"
+  type        = number
+  default     = 0
+  # Validation is done in terraform plan phase by IBM provider, so no need to add any extra validation here
+}
+
+variable "memory_mb" {
+  description = "Allocated memory per member. For more information, see https://cloud.ibm.com/docs/databases-for-redis?topic=databases-for-redis-resources-scaling"
+  type        = number
+  default     = 1024
+  # Validation is done in terraform plan phase by IBM provider, so no need to add any extra validation here
+}
+
+variable "disk_mb" {
+  description = "Allocated disk per member. For more information, see https://cloud.ibm.com/docs/databases-for-redis?topic=databases-for-redis-resources-scaling"
+  type        = number
+  default     = 1024
+  # Validation is done in terraform plan phase by IBM provider, so no need to add any extra validation here
+}
+
+variable "members" {
+  description = "Allocated number of members. Members can be scaled up but not down."
+  type        = number
+  default     = 2
+  # Validation is done in terraform plan phase by IBM provider, so no need to add any extra validation here
+}
+
+variable "endpoints" {
+  description = "Specify whether you want to enable the public, private, or both service endpoints. Supported values are 'public', 'private', or 'public-and-private'."
+  type        = string
+  default     = "private"
+  validation {
+    condition     = can(regex("public|public-and-private|private", var.endpoints))
+    error_message = "Valid values for service_endpoints are 'public', 'public-and-private', and 'private'"
+  }
+}
+
+variable "instance_name" {
+  description = "The name to give the Redis instance."
+  type        = string
+}
+
+variable "tags" {
+  type        = list(any)
+  description = "Optional list of tags to be added to the Redis instance and the associated service credentials (if creating)."
```

Comment:
Update to just say: `Optional list of tags to be added to the Redis instance` (see https://github.com/terraform-ibm-modules/terraform-ibm-icd-redis/pull/188/files#r1260832608)